product_variant_default_code/models/product.py

```diff
@@ -186,18 +158,43 @@ def _compute_default_code(self):
 class ProductProduct(models.Model):
     _inherit = "product.product"
 
-    manual_code = fields.Boolean(string="Manual Reference")
-
-    @api.model
-    def create(self, vals):
-        product = super(ProductProduct, self).create(vals)
-        if product.reference_mask:
-            render_default_code(product, product.reference_mask)
-        return product
+    manual_code = fields.Boolean(string="Manual Reference", default=False)
+    default_code = fields.Char(
+        compute="_compute_default_code", readonly=False, store=True
```

Comment:
maybe you should add an inverse method so if you fill manually the default_code it tick the manual_code to True (like the onchange_default_code